### PR TITLE
LABS-1073 Make oligarch logging separate to WAL

### PR DESCRIPTION
**Pull request by quchenhao**
@ddanderson @korteland This is mostly working except test_oligarch05. It crashes on a checksum mismatch in dir store. I think it may not because of my change. Any idea why this can fail?

Here's the stack of the failure:
```
#0  0x0000ffffa9a2f200 in ?? () from /lib/aarch64-linux-gnu/libc.so.6
#1  0x0000ffffa99ea67c in raise () from /lib/aarch64-linux-gnu/libc.so.6
#2  0x0000ffffa99d7130 in abort () from /lib/aarch64-linux-gnu/libc.so.6
#3  0x0000ffffa8e3e5e4 in __wt_abort (session=0xd6f27e8) at /home/ubuntu/wiredtiger/src/os_common/os_abort.c:30
#4  0x0000ffffa8eaffc0 in __wt_panic_func (session=0xd6f27e8, error=-31802, func=0xffffa91e1160 <__PRETTY_FUNCTION__.0> "__block_pantry_read", line=123, category=WT_VERB_DEFAULT, fmt=0xffffa91e1130 "%s: fatal read error") at /home/ubuntu/wiredtiger/src/support/err.c:570
#5  0x0000ffffa90be718 in __block_pantry_read (session=0xd6f27e8, block_pantry=0xd50ca50, buf=0xffff927cc988, block_meta=0xffff927cc470, pantry_id=3668, size=28656, checksum=1289031804) at /home/ubuntu/wiredtiger/src/block_pantry/block_read.c:123
#6  0x0000ffffa90be7a8 in __wt_bmp_read (bm=0xd51da90, session=0xd6f27e8, buf=0xffff927cc988, block_meta=0xffff927cc470, addr=0xffff927cc9bb "\251\377\377", addr_size=10) at /home/ubuntu/wiredtiger/src/block_pantry/block_read.c:145
#7  0x0000ffffa8f0db8c in __wt_blkcache_read (session=0xd6f27e8, buf=0xffff927cc988, block_meta=0xffff927cc978, addr=0xffff927cc9b1 "\316\024\342O\260\344L\324\360<\251\377\377", addr_size=10) at /home/ubuntu/wiredtiger/src/block_cache/block_io.c:122
#8  0x0000ffffa8f76ed0 in __page_read (session=0xd6f27e8, ref=0xffff7c020d30, flags=256) at /home/ubuntu/wiredtiger/src/btree/bt_read.c:186
#9  0x0000ffffa8f776a8 in __wt_page_in_func (session=0xd6f27e8, ref=0xffff7c020d30, flags=256, func=0xffffa9194200 <__PRETTY_FUNCTION__.5> "__wt_row_search", line=571) at /home/ubuntu/wiredtiger/src/btree/bt_read.c:324
#10 0x0000ffffa8fd254c in __wt_page_swap_func (session=0xd6f27e8, held=0xa6db220, want=0xffff7c020d30, flags=256, func=0xffffa9194200 <__PRETTY_FUNCTION__.5> "__wt_row_search", line=571) at /home/ubuntu/wiredtiger/src/include/btree_inline.h:2223
#11 0x0000ffffa8fd45bc in __wt_row_search (cbt=0xffff7c001ac0, srch_key=0xffff7c001be0, insert=true, leaf=0x0, leaf_safe=false, leaf_foundp=0x0) at /home/ubuntu/wiredtiger/src/btree/row_srch.c:571
#12 0x0000ffffa8f38994 in __cursor_row_search (cbt=0xffff7c001ac0, insert=true, leaf=0x0, leaf_foundp=0x0) at /home/ubuntu/wiredtiger/src/btree/bt_cursor.c:573
#13 0x0000ffffa8f3a710 in __wt_btcur_insert (cbt=0xffff7c001ac0) at /home/ubuntu/wiredtiger/src/btree/bt_cursor.c:1195
#14 0x0000ffffa8dbc0e4 in __curfile_insert (cursor=0xffff7c001ac0) at /home/ubuntu/wiredtiger/src/cursor/cur_file.c:395
#15 0x0000ffffa8d9e004 in __oligarch_log_replay_op_apply (session=0xd6f27e8, lsnp=0xffff927cde98, pp=0xffff927cd9b0, end=0xffff7c0011a8 "") at /home/ubuntu/wiredtiger/src/conn/conn_oligarch.c:619
#16 0x0000ffffa8d9e554 in __oligarch_log_replay_commit_apply (session=0xd6f27e8, lsnp=0xffff927cde98, pp=0xffff927cd9b0, end=0xffff7c001200 "\253\253\253\253\253\253\253\253\065") at /home/ubuntu/wiredtiger/src/conn/conn_oligarch.c:706
#17 0x0000ffffa8d9e83c in __oligarch_log_replay (session=0xd6f27e8, logrec=0xffff7c000b70, lsnp=0xffff927cde98, next_lsnp=0xffff927cdeb0, cookie=0x0, firstrecord=0) at /home/ubuntu/wiredtiger/src/conn/conn_oligarch.c:757
#18 0x0000ffffa8e2dc38 in __wt_oligarch_log_scan (session=0xd6f27e8, start_lsnp=0x0, end_lsnp=0x0, flags=1, func=0xffffa8d9e5a0 <__oligarch_log_replay>, cookie=0x0) at /home/ubuntu/wiredtiger/src/oligarch_log/oligarch_log.c:2419
#19 0x0000ffffa8d9ea70 in __wt_oligarch_manager_thread_run (session_shared=0xd6f27e8, thread=0xa6b2a30) at /home/ubuntu/wiredtiger/src/conn/conn_oligarch.c:796
```

With this change, we can also delete `__wt_oligarch_manager_add_table` and `__wt_oligarch_manager_remove_table` as everything in the oligarch log should be replayed.

### CODE REVIEW

File: src/txn/txn.c [hunk 1]
```diff
@@ -1839,8 +1915,10 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
          * time we release our transaction so that the LSN any checkpoint gets will always reflect
          * visible data.
          */
-        __wt_readlock(session, &txn_global->visibility_rwlock);
-        locked = true;
+        if (!locked) {
+            __wt_readlock(session, &txn_global->visibility_rwlock);
+            locked = true;
+        }
```
Comment (ddanderson) [inline src/txn/txn.c:1921]: This looks like a bug-fix rather than a mechanical change?  Is it in the original?

**Review by ddanderson — APPROVED** [review]
@quchenhao this looks great!  I spotted a couple minor things, really a nice, huge, piece of work.

File: src/txn/txn.c (hunk 1 above)
Comment (quchenhao) [reply src/txn/txn.c:1921]: No, this is not a bug fix. I need to do this because in the oligarch path, I may have locked it already.

**Review by korteland — APPROVED** [review]
LGTM, but I wasn't especially through due to time constraints.

**Pull request merged**